parser_library/test/processing/ainsert_test.cpp

```diff
@@ -28,8 +28,7 @@ TEST(ainsert, ainsert_with_substitution)
     analyzer a(input);
     a.analyze();
 
-    a.collect_diags();
-    auto& diags = a.diags();
+    auto diags = a.diags();
     ASSERT_EQ(diags.size(), 0);
```

Comment:
Can't make GH suggestion here...

```
ASSERT_TRUE(a.diags.empty());
```

Reply:
There are more of these diagnostic related things that could be upgraded. I will do a follow up PR.